### PR TITLE
arch/imx9: Add GPIO and IOMUX drivers

### PR DESCRIPTION
**Pull request by pussuw**
## Summary
This adds memory mapped registers and drivers for digital I/O and IOMUX. The clock driver is also updated to provide system PLL and root clock frequencies (some drivers need to know them).
## Impact
Add new functionality for imx9.
## Testing
imx93-evk

### CODE REVIEW

**Review by jerpelea — APPROVED** [review]

File: arch/arm64/src/imx9/imx9_clockconfig.c [hunk 1]
```diff
@@ -58,6 +58,30 @@
  * Private Data
  ****************************************************************************/
 
+static uint32_t g_pll_freqs[] =
```
Comment (xiaoxiang781216) [inline arch/arm64/src/imx9/imx9_clockconfig.c:61]: add const

File: arch/arm64/src/imx9/imx9_gpiobase.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#include "imx9_gpio.h"
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+#if defined(CONFIG_ARCH_CHIP_IMX93)
+/* Base address for the GPIO memory mapped registers */
+
+const uintptr_t g_gpio_base[] =
```
Comment (xiaoxiang781216) [inline arch/arm64/src/imx9/imx9_gpiobase.c:36]: add static

File: arch/arm64/src/imx9/imx9_gpioirq.c [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+ *
+ * Description:
+ *   Disable the interrupt for specified GPIO IRQ
+ *
+ ****************************************************************************/
+
+int imx9_gpioirq_disable(gpio_pinset_t pinset)
+{
+  uint32_t  port = (pinset & GPIO_PORT_MASK)   >> GPIO_PORT_SHIFT;
+  uint32_t  pin  = (pinset & GPIO_PIN_MASK)    >> GPIO_PIN_SHIFT;
```
Comment (xiaoxiang781216) [inline arch/arm64/src/imx9/imx9_gpioirq.c:342]: remove the extra spaces before >>

File: arch/arm64/src/imx9/imx9_gpioirq.c [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+ *
+ * Description:
+ *   Configure an interrupt for the specified GPIO pin.
+ *
+ ****************************************************************************/
+
+int imx9_gpioirq_configure(gpio_pinset_t pinset)
+{
+  uint32_t port = (pinset & GPIO_PORT_MASK)   >> GPIO_PORT_SHIFT;
+  uint32_t pin  = (pinset & GPIO_PIN_MASK)    >> GPIO_PIN_SHIFT;
```
Comment (xiaoxiang781216) [inline arch/arm64/src/imx9/imx9_gpioirq.c:274]: remove the extra spaces before >>

File: arch/arm64/src/imx9/imx9_gpioirq.c [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+ *
+ * Description:
+ *   Attach a pin interrupt handler.
+ *
+ ****************************************************************************/
+
+int imx9_gpioirq_attach(gpio_pinset_t pinset, xcpt_t isr, void *arg)
+{
+  uint32_t port = (pinset & GPIO_PORT_MASK)   >> GPIO_PORT_SHIFT;
+  uint32_t pin  = (pinset & GPIO_PIN_MASK)    >> GPIO_PIN_SHIFT;
```
Comment (xiaoxiang781216) [inline arch/arm64/src/imx9/imx9_gpioirq.c:250]: remove the extra spaces before >>

File: arch/arm64/src/imx9/imx9_clockconfig.c (hunk 1 above)
Comment (pussuw) [reply arch/arm64/src/imx9/imx9_clockconfig.c:61]: This LUT should be writable, I will add support to change PLL frequency in the future and the selected frequency will be stored in this LUT.

File: arch/arm64/src/imx9/imx9_gpioirq.c (hunk 3 above)
Comment (pussuw) [reply arch/arm64/src/imx9/imx9_gpioirq.c:342]: Sure

File: arch/arm64/src/imx9/imx9_gpio.h [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+{
+#else
+#define EXTERN extern
+#endif
+
+/* Look-up table that maps GPIO1..GPIOn indexes into GPIO register base
+ * addresses
+ */
+
+EXTERN const uintptr_t g_gpio_base[];
```
Comment (pussuw) [inline arch/arm64/src/imx9/imx9_gpio.h:226]: g_gpio_base is extern here, needed globally

File: arch/arm64/src/imx9/imx9_gpiobase.c (hunk 2 above)
Comment (pussuw) [reply arch/arm64/src/imx9/imx9_gpiobase.c:36]: This is needed globally

Comment (xiaoxiang781216) [reply arch/arm64/src/imx9/imx9_gpiobase.c:36]: Ok

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**